File: client/core/models/api/session.ts
import { FormBuilder, FormGroup, Validators } from "@angular/forms";
import { EntityBase } from "../entity-base";
import { Campaign } from "./campaign";
import { Character } from "./character";
import { SessionEvent } from "./session-event";

export interface Session extends EntityBase {
    campaignId: number;
    sessionDate: string;
    title: string;
    url: string;
    summary: string;

    campaign: Campaign;
    sessionEvents: SessionEvent[];
    characters: Character[];
}

export const GenerateSessionForm =  (session: Session, fb: FormBuilder): FormGroup => 
    fb.group({
        id: [session?.id],
        campaignId: [session.campaignId, Validators.required],
        sessionDate: [session.sessionDate, Validators.required],
        title: [session.title, Validators.required],
        url: [session.url],
        summary: [session.summary]
    })

File: server/CampaignTracker.Models/Entities/Session.cs
namespace CampaignTracker.Models;
public class Session : EntityBase
{
    public int CampaignId { get; set; }
    public string SessionDate { get; set; } // string representations of dates
    public string Title { get; set; }
    public string Url { get; set; }
    public string Summary { get; set; }

    public Campaign Campaign { get; set; }
    public IEnumerable<Event> Events { get; set; }
    public IEnumerable<Character> Characters { get; set; }
}